BookingQueueSubscriber/BookingQueueSubscriber.Services/MessageHandlers/EndpointUpdatedHandler.cs

```diff
@@ -11,26 +13,60 @@ namespace BookingQueueSubscriber.Services.MessageHandlers
     public class EndpointUpdatedHandler : IMessageHandler<EndpointUpdatedIntegrationEvent>
     {
         private readonly IVideoApiService _videoApiService;
+        private readonly ILogger<EndpointUpdatedHandler> _logger;
+        private const int RetryLimit = 3;
+        private const int RetrySleep = 3000;
 
-        public EndpointUpdatedHandler(IVideoApiService videoApiService)
+        public EndpointUpdatedHandler(IVideoApiService videoApiService, ILogger<EndpointUpdatedHandler> logger)
         {
             _videoApiService = videoApiService;
+            _logger = logger;
         }
 
         public async Task HandleAsync(EndpointUpdatedIntegrationEvent eventMessage)
         {
-            var conference = await _videoApiService.GetConferenceByHearingRefId(eventMessage.HearingId);
```

Comment:
We have a null conference object that will get populated at the start of the for loop. It will get populated within the for loop and will keep whatever value it acquired within the for loop. So even if the eventMessage.DefenceAdvocate is null we will still have the conference value we acquired before that.

Reply:
But the conference will only be fetched if eventMessage.DefenceAdvocate is not null or empty. If it is null or empty, the conference won't be fetched and the endpoint won't be updated. Unlike the existing behaviour, which is to update the endpoint regardless of the value of eventMessage.DefenceAdvocate.

Reply:
conference with be fetched before entering into the retry loop to get the defence advocate. The retry loop shall be exited should the conference ever be null. At the end of the retry loop the conference shall be acquired again. This is in order to get the Participants list, but the conference should never be null.